[WIP] Add connection status to environment card

rottencandy opened the pull request:
## Fixes 
https://issues.redhat.com/browse/HAC-3599
<!-- For e.g Fixes: https://issues.redhat.com/browse/HAC-XXX -->


## Description
Add connection status to environment card using GitOpsDeploymentManagedEnvironment and update card description.

Depends on #562 , only consider commit c816dd44b2ca8b8d367769e205645c90ab501926
<!-- Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change. -->


## Type of change
<!-- Please delete options that are not relevant. -->

- [x] Feature
- [ ] Bugfix
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Screen shots / Gifs for design review 
![0](https://user-images.githubusercontent.com/20013884/232137961-6634b884-5d70-4cf8-acb5-d3c1d6413eea.png)
![1](https://user-images.githubusercontent.com/20013884/232137972-7a2e075f-0621-4b2c-92d5-5f7af6be7a26.png)
![2](https://user-images.githubusercontent.com/20013884/232137982-8ec79139-65db-4e9a-8f46-073142c0413b.png)
<!-- If change affects UI in any way, tag relevant UX people and add screenshots/gifs  -->


<!-- Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration -->

<!-- - [ ] Test A -->
<!-- - [ ] Test B -->

<!-- **Test Configuration(s)**: -->


## Browser conformance: 
<!-- To mark tested browsers, use [x] -->
- [ ] Chrome
- [x] Firefox
- [ ] Safari
- [ ] Edge


<!-- ## Checklist: -->

<!-- 
- [ ] Code follows the style guidelines
- [ ] Self-reviewed the code
- [ ] Added comments in hard-to-understand areas
- [ ] Made corresponding changes to the documentation
- [ ] Changes generate no new warnings
- [ ] Added tests that prove this fix is effective or that the feature works
- [ ] New and existing unit tests pass locally with new changes
- [ ] Any dependent changes have been merged and published in downstream modules 
-->

rottencandy commented:
/cc @pankajshivpuje @rohitkrai03

pankajshivpuje commented:
@rottencandy We need to update the environment card to show cluster type and remove application status and last deploy. Only label we are using is for type values. All the other text should be normal (no labels).

rottencandy commented:
Removed labels and app status fields.

Since the CR doesn't yet provide the cluster type, I've added a hardcoded OpenShift type for now. @rohitkrai03 is this fine?
![0](https://user-images.githubusercontent.com/20013884/232844558-b094afb1-133f-4e93-bdec-57751c1364a4.png)

I've used the default color for the alert message at the top, since it's more consistent with existing alerts in UI, but I can change it if needed.

rohitkrai03 commented:
@rottencandy I don't think we should remove application status from environments card in application details page yet. It's still useful information I think. @pankajshivpuje WDYT?

pankajshivpuje commented:
> @rottencandy I don't think we should remove application status from environments card in application details page yet. It's still useful information I think. @pankajshivpuje WDYT?

My context was more of environment cards in the environment tab. If there are multiple application deployed on that environment, what application the status points to? I am not aware of how application status is used on application details page. @Ranelim Any thoughts?

rohitkrai03 commented:
> > @rottencandy I don't think we should remove application status from environments card in application details page yet. It's still useful information I think. @pankajshivpuje WDYT?
> 
> My context was more of environment cards in the environment tab. If there are multiple application deployed on that environment, what application the status points to? I am not aware of how application status is used on application details page. @Ranelim Any thoughts?

@pankajshivpuje We don't show application status on environments list page. We only show application status if you're in the context of application details view.

rottencandy commented:
Environments page:
![0](https://user-images.githubusercontent.com/20013884/233163880-b3a73b2d-f7e9-4599-9018-3679b4cdd0f6.png)
Application details page:
![1](https://user-images.githubusercontent.com/20013884/233163941-d70656da-8ef4-406c-96bf-586119fb521e.png)

rohitkrai03 commented on src/components/Environment/EnvironmentCard.tsx:
```diff
@@ -94,15 +103,32 @@ const EnvironmentCard: React.FC<EnvironmentCardProps> = ({ environment, readOnly
             environment={environment as EnvironmentKindWithHealthStatus}
           />
+
+          <DescriptionListGroup>
```
This can go before the application status. Having application status info at the end keeps it consistent for both views.

rohitkrai03 commented:
/retest

rohitkrai03 reviewed (approved):
/lgtm
